src/main/kotlin/no/nav/syfo/client/dokarkiv/domain/Dokumentvariant.kt

```diff
@@ -0,0 +1,20 @@
+package no.nav.syfo.client.dokarkiv.domain
+
+enum class filtypeType(
+    val value: String,
+) {
+    PDFA("PDFA"),
+}
+
+enum class variantformatType(
+    val value: String,
+) {
+    ARKIV("ARKIV"),
+}
+
+data class Dokumentvariant(
+    val filnavn: String? = null, // MAX 200 characters
```

Comment:
Bør det valideres i konstruktøren så man ikke får opprettet Dokumentvariant med for langt filnavn?

Reply:
Og hvordan sikrer man at variantformat som sendes inn her er en gyldig variantformatType når den kommer inn som string? Samme gjelder alle steder man sender inn enum-verdier som string i dokarkiv-klassene. Man kan jo vurdere å sende inn en enum-typer i stedet og sette til string-verdien i konstruktøren.

Reply:
1. Lagt til validering for lengde nå,
2. Det kan fungere bra det. Prøver på å få det til nå uten å måtte gå vekk fra "data class" og måtte implementere enn full data class selv. Eneste måten jeg har fått det til foreløpig er å ta input som string men med validering som sjekker at string tilsvarer en gyldig enum.

Reply:
Ja, vi vil jo ha data class. Mulig jackson kan løse det automagisk eller om man kan bruke noe jackson-annotasjon på feltene for å få det til ja.  Alternativ kan man lage factory-metoder for å opprette disse objektene med enum som param og så kan den ta seg av å instansiere data-klasse ala forslaget her: https://stackoverflow.com/a/52095214/15398102, men det blir kanskje litt styr 🤔

Reply:
Har endret til bruk av en variant av eksempelet med companion-factory som du lenket til og hvor man sender inn Enum-klasse i stedet for string.

Reply:
Ser bra ut 👏 ✔️